nixos/tests/prometheus-exporters.nix

```diff
@@ -257,6 +257,47 @@ let
       '';
     };
 
+    exportarr-sonarr = {
+      nodeName = "exportarr_sonarr";
+      exporterConfig = {
+        enable = true;
+        url = "http://127.0.0.1:8989";
+        # testing for real data is tricky, because the api key can not be preconfigured
+        apiKeyFile = pkgs.writeText "dummy-api-key" "eccff6a992bc2e4b88e46d064b26bb4e";
+      };
+      metricProvider = {
+        services.sonarr.enable = true;
+        systemd.services.sonarr.serviceConfig.ExecStartPre =
+          let
+            sonarr_config = pkgs.writeText "config.xml" ''
```

Comment:
Drive-by comment: it might be nice to make (some) of those configuration options part of the *arr modules? I'm mainly thinking about the port(s), log level, bind address, and API key.

Though I'm not sure how the declarative options and web-ui options would mesh.

Feel free to dismiss this, I'm just curious as I never really looked into the config files for this suite of software.

Reply:
that is a nice idea. I would suggest to have this in a separate MR. might be relevant for multiple *arr services.